pkg/operator/staticpod/controllers.go

```diff
@@ -51,6 +51,7 @@ type staticPodOperatorControllerBuilder struct {
 	installCommand           []string
 	installerPodMutationFunc installer.InstallerPodMutationFunc
 	minReadyDuration         time.Duration
+	deployStartupMonitor     func() (bool, error)
```

Comment:
why not `enableStartupMonitorFn func() bool` ?

Reply:
oh, we are going to read from the infra lister, so we need an `error` as well, right?

Reply:
yes, you told me :D

Reply:
:) yeah, we need an `error` as well.

I would rename it to `startupMonitorEnabled func() (bool, error)` to align with the name we have in the controller.

Reply:
done